### PR TITLE
Use GitHub Issue Form for Bug Report Template

### PR DESCRIPTION
**Pull request by TanayParikh**
Leverage [GitHub issue form templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) to provide an improved bug creation experience for community contributors.

[Preview](https://github.com/dotnet/aspnetcore/blob/TanayParikh-patch-5/.github/ISSUE_TEMPLATE/10_bug_report.yml)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/10_bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    label: Exceptions (if any)
+    description: Include the exception you get when facing this issue.
+    placeholder: 
+  validations:
+    required: false
+- type: input
+  attributes:
+    label: .NET Version 
+    description: |
+      Run `dotnet --version`
```
Comment (Tratcher) [inline .github/ISSUE_TEMPLATE/10_bug_report.yml:59]: Why dotnet --version vs dotnet --info?

File: .github/ISSUE_TEMPLATE/10_bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      More information on our issue management policies can be found here: https://aka.ms/aspnet/issue-policies
+
+      Please keep in mind that the GitHub issue tracker is not intended as a general support forum, but for reporting **non-security** bugs and feature requests.
+
+      If you believe you have an issue that affects the SECURITY of the platform, please do NOT create an issue and instead email your issue details to secure@microsoft.com. Your report may be eligible for our [bug bounty](https://www.microsoft.com/en-us/msrc/bounty-dot-net-core) but ONLY if it is reported through email.
+      For other types of questions, consider using [StackOverflow](https://stackoverflow.com).
+- type: checkboxes
+  attributes:
+    label: Is there an existing issue for this?
+    description: Please search to see if an issue already exists for the bug you encountered.
```
Comment (Tratcher) [inline .github/ISSUE_TEMPLATE/10_bug_report.yml:16]: Link to /issues?

**Review by Tratcher — APPROVED** [review]

**Review by Tratcher — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/10_bug_report.yml (hunk 1 above)
Comment (TanayParikh) [reply .github/ISSUE_TEMPLATE/10_bug_report.yml:59]: I have `dotnet --info` in the next field. I could just remove this `input` field, but I had added it as `dotnet --version` is usually the most pertinent information for us. 

In the current template we're able to put `dotnet --info` within a `<details>` section so it doesn't take a bunch of vertical space in the issue (with the full `dotnet --info` output you have to scroll a decent amount to get to the comments). With this new template `<details>` doesn't appear to be possible. So we may get a lot of issues that have 100-200 lines of `dotnet --info` related output.

**Review by rafikiassumani-msft — APPROVED** [review]

**Review by captainsafia — APPROVED** [review]
Great addition! Thanks for adding this.

**Comment by TanayParikh** [comment]
Merged & validated: https://github.com/dotnet/aspnetcore/issues/39066

**Comment by halter73** [comment]
Is there a way to see a preview of what the issue will look like before submitting it when using an issue form?

**Comment by TanayParikh** [comment]
> Is there a way to see a preview of what the issue will look like before submitting it when using an issue form?

Doesn't seem to be possible 😕

**Comment by dougbu** [comment]
> > Is there a way to see a preview of what the issue will look like before submitting it when using an issue form?

> Doesn't seem to be possible 😕

Looks like GitHub automagically displays the finished form if you use "View file" on the YAML instead of perusing the whole change 😺

See https://github.com/dotnet/aspnetcore/blob/4b1b75bc4bce29cdc90321836fd90ab01f727e32/.github/ISSUE_TEMPLATE/10_bug_report.yml for example

**Pull request merged**